server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -391,29 +393,36 @@ DateTimeZone rewriteTimeZone(QueryShardContext context) throws IOException {
                 }
 
                 if (anyInstant != null) {
-                    final long prevTransition = tz.previousTransition(anyInstant);
-                    final long nextTransition = tz.nextTransition(anyInstant);
+                    Instant instant = Instant.ofEpochMilli(anyInstant);
+                    final long prevTransition = tz.getRules().previousTransition(instant).getInstant().toEpochMilli();
```

Comment:
Can we guarantee that there is always a previous transition in our case? I am asking because that method might actually return `null` in that case. In my understanding the earliest instant that we support is 0 (in epoch millis) and intuitevely I'd expect that there is a previous transition for any time zone but then time zones are not necessarily intuitive... .

Reply:
good catch!